source/plugin/offline-userplayer-data.rst

```diff
@@ -0,0 +1,71 @@
+===================
+Offline Player Data
+===================
+
+It may be necessary for plugins to access player data even when the player is offline.
+One could guess that ``Sponge.getServer().getPlayer()`` returning a ``Player`` can be used for this.
+But since ``Player`` objects only exist for online players, another solution has to be used.
+
+Some plugins store the relevant data themselves and associate the user by using the ``GameProfileManager``.
+But writing different code for offline and online users is not necessary:
+The ``ServiceManager`` natively provides a service known as the ``UserStorageService`` which is capable of returning ``User`` instances for ``Player``\s who are currently offline.
+Since the ``Player`` interface extends ``User`` most methods you call on a ``Player`` are also available. 
+
+For example:
+
+* ``#hasPermission(String permission)`` is available from both instances.
+
+Code Example
+----------------------------------
```

Comment:
Why so many dashes here? They should match up with the lining of 'Code Example'.